libpod/image_inspect.go

```diff
@@ -0,0 +1,81 @@
+package libpod
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/containers/image/types"
+	"github.com/containers/storage"
+	digest "github.com/opencontainers/go-digest"
+	ociv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/libpod/driver"
+)
+
+func getImageData(img storage.Image, imgRef types.Image, size int64, driver *driver.Data) (*ImageData, error) {
+	imgSize, err := imgRef.Size()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading size of image %q", img.ID)
+	}
+	manifest, manifestType, err := imgRef.Manifest()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading manifest for image %q", img.ID)
+	}
+	imgDigest := digest.Digest("")
+	if len(manifest) > 0 {
+		imgDigest = digest.Canonical.FromBytes(manifest)
+	}
+	annotations := annotations(manifest, manifestType)
+
+	ociv1Img, err := imgRef.OCIConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error gettin oci image info %q", img.ID)
+	}
+	info, err := imgRef.Inspect()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting image info %q", img.ID)
+	}
+
+	var repoDigests []string
+	for _, name := range img.Names {
+		repoDigests = append(repoDigests, strings.SplitN(name, ":", 2)[0]+"@"+imgDigest.String())
+	}
+
+	data := &ImageData{
+		Id:            img.ID,
+		RepoTags:      img.Names,
+		RepoDigests:   repoDigests,
+		Comment:       ociv1Img.History[0].Comment,
+		Created:       ociv1Img.Created,
+		Author:        ociv1Img.History[0].Author,
+		Architecture:  ociv1Img.Architecture,
+		Os:            ociv1Img.OS,
+		Config:        &ociv1Img.Config,
+		DockerVersion: info.DockerVersion,
```

Comment:
I think we should rename this, exact compatability be damned. @rhatdan agree?

Reply:
I agree.  We don't need to be outputing things that say Docker

Reply:
rename it to what?  OCIVersion?

Reply:
Personally I'd just drop the 'Docker' part and leave it as Version:

Reply:
but it should still get its data from info.DockerVersion?

Reply:
I'd rename that too.

Reply:
👍 Version works for me.

Reply:
fixed